egs/wsj/s5/utils/lang/limit_arpa_unk_history.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+
+
+# Copyright 2018    Armin Oliya
+# Apache 2.0.
+
+
+# This script takes an existing ARPA lanugage model and limits the <unk> history 
+# to make it suitable for downstream <unk> modeling. This is for the case when
+# you don't have access to the original text corpus that is used for creating the LM.
+# If you do, you can use pocolm with the option --limit-unk-history=true. This
+# keeps the graph compact after adding the unk model.
+
+
+
+import argparse
+import codecs 
+import os
+import re 
+from collections import defaultdict
+
+
+
+parser = argparse.ArgumentParser(description='This script takes an existing ARPA lanugage model and limits the <unk> history to make it suitable for downstream <unk> modeling.')
+parser.add_argument('--old_lm',required=True, help='path to the old ARPA language model', type=str)
```

Comment:
I think it would be better to have this program read from stdin and write to stdout, because these files are usually stored gzipped.  You should then make any informational statements be printed to stderr.